docs/site/reference/error-codes.md

```diff
@@ -26,3 +27,70 @@ See the following resources for a list of low-level error codes:
 
 - [Common System Errors](https://nodejs.org/api/errors.html#errors_common_system_errors)
 - [Node.js Error Codes](https://nodejs.org/api/errors.html#errors_node_js_error_codes)
+
+## CANNOT_INFER_PROPERTY_TYPE
+
+LoopBack is using TypeScript design-time type metadata to automatically infer
+simple property types like `string`, `number` or a model class.
+
+In the following example, the type of the property `name` is inferred from
+TypeScript metadata as `string`.
+
+```ts
+@model()
+class Product {
+  @property()
+  name: string;
+}
+```
+
+Design-time property type is not available in the following cases:
+
+- The property has a type not supported by TypeScript metadata engine:
+  - `undefined`
+  - `null`
+  - complex types like arrays (`string[]`), generic types (`Partial<MyModel>`),
+    union types (`string | number`), and more.
+- The TypeScript project has not enabled the compiler option
+  `emitDecoratorMetadata`.
+- The code is written in vanilla JavaScript.
+
+<a id="cannot_infer_property_type-solutions"></a>
```

Comment:
This html snippet creates a unique anchor. I think `### Solutions` creates the same anchor `#solutions` for all error codes, which is not useful.